contracts/tips/src/utils.rs

```diff
@@ -0,0 +1,75 @@
+use cosmwasm_std::Coin;
+use std::collections::btree_map::BTreeMap;
+
+/// Iterates over the coins vector and merges the coins having the same `denom`.
+/// * `coins` - Vector of coins to merge.
+pub fn merge_coins(coins: Vec<Coin>) -> Vec<Coin> {
+    if coins.len() <= 1 {
+        return coins;
+    }
+
+    let mut map: BTreeMap<String, u128> = BTreeMap::new();
+    for coin in coins {
+        let value = map.get_mut(&coin.denom);
+
+        if let Some(amount) = value {
+            *amount += coin.amount.u128();
```

Comment:
Should we use the safe math Cosmwasm provide instead the raw add?
https://docs.rs/cosmwasm-std/0.14.0-rc1/cosmwasm_std/struct.Uint128.html#method.checked_add